### PR TITLE
Added contributer guide to docs

### PR DESCRIPTION
**Pull request by jbusecke**

### CODE REVIEW

File: doc/contributer_guide.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+All conda environments contain black and you can reformat code using::
+
+    black xgcm
+
+`pre-commit <https://pre-commit.com/>`_ provides an automated way to reformat your code
+prior to each commit. Simply install pre-commit::
+
+    pip install pre-commit
+
+and your code will be properly formatted before each commit.
```
Comment (rabernat) [inline doc/contributer_guide.rst]: You also need to run `pre-commit install` in the xgcm directory before it works.

Comment (jbusecke) [reply doc/contributer_guide.rst]: Ah yes. Thanks for catching that.

File: doc/contributer_guide.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.. _contributer_guide:
+
+Contributer Guide
```
Comment (rabernat) [inline doc/contributer_guide.rst]: It's spelled **Contributor** 

Could you change throughout?

**Comment by rabernat** [comment]
I think this is a fine start! Once the spelling error is fixed I will merge.

File: doc/contributer_guide.rst (hunk 2 above)
Comment (jbusecke) [reply doc/contributer_guide.rst]: oh sorry. of course

**Pull request merged**